test/version.jl

```diff
@@ -94,8 +94,7 @@ show(io,v"4.3.2+1.a")
 
 # typemin and typemax
 @test typemin(VersionNumber) == v"0-"
-@test typemax(VersionNumber) ==
-  VersionNumber(typemax(Int), typemax(Int), typemax(Int), (), ("",))
```

Comment:
Sure, but that's not directly tested either from what I can tell. In the unlikely event that some bug or mistake happened to make `typemax(VersionNumber)` start outputting, say, `v"5.0.0"`, would be good to see an obvious test failure from it.